crates/p2p/src/swarm/mod.rs

```diff
@@ -56,17 +56,31 @@ static TOPIC: LazyLock<Sha256Topic> = LazyLock::new(|| Sha256Topic::new("bitvm2"
 /// Global name of the protocol
 const PROTOCOL_NAME: &str = "/strata-bitvm2";
 
-#[derive(Snafu, Debug)]
+#[derive(Debug, Error)]
 pub enum Error {
-    #[snafu(display("Database error: {source}"))]
-    Repository { source: RepositoryError },
-
-    #[snafu(whatever, display("{message}"))]
-    Whatever {
-        message: String,
-        #[snafu(source(from(Box<dyn std::error::Error>, Some)))]
-        source: Option<Box<dyn std::error::Error>>,
+    #[error("Database error: {source}")]
+    Repository {
+        #[source]
+        source: RepositoryError,
     },
```

Comment:
Isn't this one that should work? Or does it conflict with others?

```suggestion
    #[error("Database error: {source}")]
    Repository(#[from] RepositoryError)
```